datafusion/core/src/physical_optimizer/enforce_distribution.rs

```diff
@@ -269,11 +266,18 @@ impl PhysicalOptimizerRule for EnforceDistribution {
 /// 4) If the current plan is Projection, transform the requirements to the columns before the Projection and push down requirements
 /// 5) For other types of operators, by default, pushdown the parent requirements to children.
 ///
+type RequiredKeyOrdering = Vec<Arc<dyn PhysicalExpr>>;
```

Comment:
It might make this code simpler to make this `struct RequiredKeyOrder` and give its methods names rather than dealing with `Vec<Vec<...>>` -- but that can be a refactor for a different PR perhaps